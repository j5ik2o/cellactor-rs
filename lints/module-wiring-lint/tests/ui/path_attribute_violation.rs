#![warn(no_parent_reexport)]

#[path = "path_attribute_violation/leaf.module"]
mod leaf;

fn use_leaf() -> leaf::Leaf {
  leaf::Leaf
}

fn main() {
  let _ = use_leaf();
}
