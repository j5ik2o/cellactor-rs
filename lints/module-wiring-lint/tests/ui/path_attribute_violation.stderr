warning: モジュール宣言で `#[path = "..."]` 属性は禁止されています
  --> $DIR/path_attribute_violation.rs:3:1
   |
LL | #[path = "path_attribute_violation/leaf.module"]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: ルール: 原則ルール: モジュールはファイル配置と `mod` 宣言だけで結線してください
   = note: 詳細: 属性 `#[path = "path_attribute_violation/leaf.module"]` が設定されています
   = help: モジュールファイルを規約どおりの場所に配置し、`mod foo;` として宣言してください
note: the lint level is defined here
  --> $DIR/path_attribute_violation.rs:1:9
   |
LL | #![warn(no_parent_reexport)]
   |         ^^^^^^^^^^^^^^^^^^

warning: 1 warning emitted

